### ISSUE TITLE
Support building on macos-arm

### ISSUE THREAD

**Issue by Miepee** [randovania/randomprime#32]
GH's macos runner switched to arm by default. 
x86_64 is still supported for the time being, but we should really try to figure out what stops it from compiling and fix it

Example log: https://github.com/randovania/randovania/actions/runs/8899395603/job/24438949675

**Comment by toasterparty** [comment]
@Miepee fixed by [py-randomprime #50](https://github.com/randovania/py-randomprime/pull/50), yeah?

**Comment by Miepee** [comment]
no. That PR just made everything consistent to macos-13, which is an x86_64 runner.

**Comment by Miepee** [comment]
macos-14 is arm only. So our deadline is basically the deprecation of macos-13.

**Comment by henriquegemignani** [comment]
It's less macos-13 being dropped and more needing to straight up support arm. I expect macos-13 to be dropped only after all supported mac hardware being arm-only. (as in, I bet it's getting longer support)

**Comment by Miepee** [comment]
Should be fixed by https://github.com/randovania/py-randomprime/pull/64
